src/lib/DepositController.js

```diff
@@ -220,4 +224,54 @@ export class DepositController {
     const importFilesUrl = draft.links.self + '/actions/files-import';
     await this.fileUploader.importParentRecordFiles(importFilesUrl, { store });
   }
+
+  /**
+   * Reserve a PID
+   */
+  reservePID = async (draft, pidType, { formik, store }) => {
+    const recordSerializer = store.config.recordSerializer;
+
+    const response = await this.apiClient.reservePID(draft, pidType);
+
+    let data = recordSerializer.deserialize(response.data || {});
+    let errors = recordSerializer.deserializeErrors(response.errors || []);
+
+    if (200 <= response.code && response.code < 300 && _isEmpty(errors)) {
```

Comment:
doubt: arent we just returning 200? why is the range needed?

Reply:
Honestly, this should very much refactored, it is a copy/paste as above to keep things the same and refactor all of them at once in the right way.